AutoCollection/HttpRequestParser.ts

```diff
@@ -196,6 +213,32 @@ class HttpRequestParser extends RequestParser {
         return value;
     }
 
+    private parseW3CTraceHeaders(request: http.IncomingMessage, requestId?: string) {
+        const tracestateHeader = request.headers[RequestResponseHeaders.traceStateHeader];
+        const traceparentHeader = request.headers[RequestResponseHeaders.traceparentHeader];
+        const requestIdHeader = request.headers[RequestResponseHeaders.requestIdHeader];
+
+
+        this.traceparent = new TraceParent(traceparentHeader, requestIdHeader);
+        this.tracestate = new TraceState(tracestateHeader);
+
+        // If !traceparent && tracestate, expectation is to discard tracestate and generate new traceparent
+        if (!traceparentHeader && tracestateHeader) {
+            this.tracestate = null;
+        }
+        this.operationId = this.traceparent.traceId;
```

Comment:
I suggest that we separate/decouple W3C TraceContext and legacy AI operation compat, instead of putting compat logic inside W3C implementation.

Reply:
Refactored to separate each parsing scenario